pkg/components/modelserver/modelserver.go

```diff
@@ -216,35 +219,47 @@ func NewConfigMap(d components.Detail, k *v1alpha1.Kepler) *corev1.ConfigMap {
 			Namespace: components.Namespace,
 			Labels:    labels,
 		},
-		Data: msConfig.Merge(trainerSettings).ToMap(),
+		Data: msConfig.ToMap(),
 	}
 }
 
 func settingsForTrainer(trainer *v1alpha1.ModelServerTrainerSpec) k8s.StringMap {
-	if trainer == nil {
-		return nil
-	}
+	msConfig := k8s.StringMap{}
+
+	prom := trainer.Prom
+	if prom != nil {
```

Comment:
(nit): prefer early returns for better readability 

```suggestion
	if prom == nil {
	   return msConfig
	}
```

Reply:
I intentionally not return it at the beginning for the future case when we have more setting to the trainer not only prom.

Reply:
Please feel free to ignore `nits`. 
imho, it is better to write the code for the current scenario and readjust if/when there is a change in future 👼 (YAGNI) :)